iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2504,4 +2504,190 @@ TYPED_TEST(stringTyped_test, AccessMaxPositionOfNotEmptyStringViaConstSubscriptO
+{
+    ::testing::Test::RecordProperty("TEST_ID", "91f733e0-5ded-4f85-9727-241b649bd7dc");
+    string<10> sut("Hypno");
+    ASSERT_TRUE(sut.insert(5, "toad", 4));
+    EXPECT_THAT(sut.size(), Eq(9));
+    EXPECT_THAT(sut.c_str(), StrEq("Hypnotoad"));
+}
+
+TYPED_TEST(stringTyped_test, InsertStringLiteralToEmptyStringWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "eba320fc-9909-4043-a944-7a5c9f2a4272");
+    ASSERT_TRUE(this->testSubject.insert(0, "M", 1));
+    EXPECT_THAT(this->testSubject.size(), Eq(1));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq("M"));
+}
+
+TYPED_TEST(stringTyped_test, InsertEmptyStringLiteralDoesNotChangeTheString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "37d723b7-5b35-4f1d-ac33-23dd483121a1");
+    this->testSubject = "M";
+    ASSERT_TRUE(this->testSubject.insert(0, "", 0));
+    EXPECT_THAT(this->testSubject.size(), Eq(1));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq("M"));
+}
+
+TYPED_TEST(stringTyped_test, InsertStringLiteralWithCountGreaterThanSizeOfStringLiteralFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3d2951f8-d868-4998-bb33-58424aba7a65");
+    ASSERT_FALSE(this->testSubject.insert(0, "M", 2));
+    EXPECT_THAT(this->testSubject.size(), Eq(0));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq(""));
+}
+
+TEST(String10, InsertTooLargeStringLiteralFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "0f11f387-f437-4a39-8207-259df4807c70");
+    string<10> sut("Ferdinand");
+    ASSERT_FALSE(sut.insert(9, "Spitzschnueffler", 16));
+    EXPECT_THAT(sut.size(), Eq(9));
+    EXPECT_THAT(sut.c_str(), StrEq("Ferdinand"));
+}
+
+TEST(String10, InsertTooLargeStringLiteralWithSmallCountSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2763b3d8-3023-4841-80d1-798acd3b71d7");
+    string<10> sut("Ferdinand");
+    ASSERT_TRUE(sut.insert(9, "Spitzschnueffler", 1));
+    EXPECT_THAT(sut.size(), Eq(10));
+    EXPECT_THAT(sut.c_str(), StrEq("FerdinandS"));
+}
+
+TYPED_TEST(stringTyped_test, InsertStringLiteralWithCount0DoesntChangeTheString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "90e80302-1c14-48bb-b9e9-0f8fca017fea");
+    ASSERT_TRUE(this->testSubject.insert(0, "Ferdinand", 0));
+    EXPECT_THAT(this->testSubject.size(), Eq(0));
+    EXPECT_THAT(this->testSubject.c_str(), StrEq(""));
+}
+
+TEST(String10, InsertStringLiteralAtPositionGreaterStringSizeFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a3a30ea5-f228-48f4-9497-08502e4f2c9a");
+    string<10> sut("Muesli");
+    ASSERT_FALSE(sut.insert(7, "s", 1));
+    EXPECT_THAT(sut.size(), Eq(6));
+    EXPECT_THAT(sut.c_str(), StrEq("Muesli"));
+}
+
+TEST(String10, InsertCxxStringAtTheBeginningOfTheStringSucceeds)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "df62df30-0838-4282-91f7-b39b1c861861");
+    string<10> sut("toad");
+    string<10> string_to_insert("Hypno");
+    ASSERT_TRUE(sut.insert(0, string_to_insert, 5));
+    EXPECT_THAT(sut.size(), Eq(9));
```

Comment:
Why not?

```suggestion
    EXPECT_THAT(sut.size(), Eq(sut.size() + string_to_insert.size()));
```

Reply:
Because `sut.size()` has already changed. But I will add an `expectedString` as you proposed above.